protos/flyteidl/event/event.proto

```diff
@@ -150,4 +150,20 @@ message TaskExecutionEvent {
     // that should be recorded regardless of the lack of phase change.
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
+
+    // Metadata around how a task was executed.
+    TaskExecutionMetadata metadata        = 16;
+}
+
+// Holds metadata around how a task was executed.
+message TaskExecutionMetadata {
+    // Includes the type of machine used for this specific task execution. 
+    enum InstanceType {
+        // The default instance type configured for the flyte application platform.
+        DEFAULT = 0; 
+
+        // The instance type configured for interruptible tasks.
+        INTERRUPTIBLE = 1;
+    }
+    InstanceType instance_type = 1;
```

Comment:
can it ever run on both?  if it gets interrupted three time and the fourth time we use a non-spot?

Reply:
yep it can, but that's why this is per task execution. if we retry on a non-spot node that retry attempt will be a distinct task execution and the instance type will change for its metadata